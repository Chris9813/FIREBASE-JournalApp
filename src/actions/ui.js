import { types } from "../types/types";

export const setError = (err) => ({
    type: types.uiSetError,
    payload: err,
})

export const remError = (err) => ({
    type: types.uiRemoveError,
})

