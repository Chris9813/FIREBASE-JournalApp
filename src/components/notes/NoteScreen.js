import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { activeNote, startDeleting } from '../../actions/notes'
import { useForm } from '../hooks/useForm'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
    const dispatch = useDispatch();

    const {active:note} = useSelector(state => state.notes)
    const [values, handleInputChange, reset] = useForm(note)
    
    const {body, title, id} = values

    const activeId = useRef( note.id )

    useEffect(() => {
        if(note.id !== activeId.current){
            reset(note)
            activeId.current = note.id
        }
    }, [note, reset])

    useEffect(() => {
        dispatch(activeNote(values.id, {...values}))
        
    }, [values, dispatch])

    const handleDelete = () => {
        dispatch(startDeleting(id));
    }

    return (
        <div className="notes_main-content">
            <NotesAppBar />

            <div className="notes_content">
                <input 
                type="text"
                placeholder="Another awsome title"
                className="notes_title-input"
                autoComplete="off"
                value = {title}
                onChange = {handleInputChange}
                name="title"
                />

                <textarea
                placeholder="What happened today"
                className="notes_textarea"
                value={body}
                onChange={handleInputChange}
                name="body"
                >
                </textarea>

                {
                note.url &&
                <div className="notes_image">
                    <img 
                    src={note.url}
                    alt="Imagen"
                    height= "150px"
                    />
                </div>
                }
            </div>
            <button className="btn btn-danger"
            onClick={handleDelete}
            >
                    Delete
            </button>
        </div>
    )
}
