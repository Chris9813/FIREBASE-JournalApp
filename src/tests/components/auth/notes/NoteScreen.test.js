import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, mount } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { activeNote } from "../../../../actions/notes";
import { NoteScreen } from "../../../../components/notes/NoteScreen";


jest.mock("../../../../actions/notes", () => ({
    activeNote: jest.fn(),
}));


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

configure({ adapter: new Adapter() });

const initState = {
    auth: {name: 'chris'},
    ui: {
        loading: false,
        msgError: null,
    },
    notes: {
        active:{
            id: 1234,
            title: 'hola',
            body: 'mundo',
            date: 0,

        },
        notes: [],
    }
};

let store = mockStore(initState)
store.dispatch = jest.fn();

describe('Pruebas en <NoteScreen />', () => {
    
    beforeEach(() => {
        store = mockStore(initState);
        jest.clearAllMocks();
    });

    const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <NoteScreen />
        </MemoryRouter>
    </Provider>
    );

    test('should ', () => {
        expect( wrapper ).toMatchSnapshot();

    });
    
    test('debe de disparar el activeNOte ', () => {
        wrapper.find('input[name="title"]').simulate('change', {
            target: {
                name: 'title',
                value: 'holalala'
            }
        });

        expect( activeNote ).toHaveBeenLastCalledWith(
            1234,
            {
                body: 'mundo',
                title: 'holalala',
                id: 1234,
                date: 0,
            }
        );

    });
    

})
