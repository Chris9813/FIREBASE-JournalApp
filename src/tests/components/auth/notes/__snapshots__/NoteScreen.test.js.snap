// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NoteScreen /> should  1`] = `
<div
  className="notes_main-content"
>
  <div
    className="notes_appbar"
  >
    <span>
      13 de agosto 1998
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes_content"
  >
    <input
      autoComplete="off"
      className="notes_title-input"
      name="title"
      onChange={[Function]}
      placeholder="Another awsome title"
      type="text"
      value="hola"
    />
    <textarea
      className="notes_textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today"
      value="mundo"
    />
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
